### PR TITLE
Use a RFID reader to check in/out of the space

### PR DESCRIPTION
**Pull request by c1ko**

### CODE REVIEW

File: mac_kicker.py [hunk 1]
```diff
@@ -16,38 +16,87 @@ def mac_tester():
...
+
+def rfid_watcher():
+	global current_rfid_users
+	rfid_reader = evdev.InputDevice('/dev/input/event0')
+	print("Connected to RFID Reader")
+	current_code = ""
+	keys = "XX1234567890XXXXqwertzuiopXXXXasdfghjklXXXXXyxcvbnmXXXXXXXXXXXXXXXXXXXXXXX"
+
+	for event in rfid_reader.read_loop():
+		if event.type == 1 and event.value == 1:
```
Comment (lynxis) [inline mac_kicker.py]: what is event.type == 1 und event.value == 1?

File: mac_kicker.py [hunk 2]
```diff
@@ -16,38 +16,87 @@ def mac_tester():
...
+			if event.code > len(keys):
```
Comment (lynxis) [inline mac_kicker.py:69]: why? (add comment)

File: rfid_reader.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python3
+import evdev, requests, hashlib
```
Comment (lynxis) [inline rfid_reader.py:2]: requests is unused.

**Comment by Kunsi** [comment]
Mir fehlt da Feedback, ob man sich gerade ein- oder ausgeloggt hat per RFID ...

File: mac_kicker.py [hunk 4]
```diff
@@ -2,10 +2,10 @@
 #
 # MIT
 # ciko@afra-berlin.de
-import pydle, threading, subprocess, time, asyncio
+import pydle, threading, subprocess, time, asyncio, hashlib, requests, evdev
```
Comment (lynxis) [inline mac_kicker.py]: Usually there is only one import per line. Also please sort the import. First the stdlibary, then the external libraries.
std library: time, hashlib, threading, subprocess, asyncio
external library: requests, pydle, requests, evdev

File: mac_kicker.py [hunk 5]
```diff
@@ -16,38 +16,87 @@ def mac_tester():
...
+		if line.split()[0].upper() == hashlib.sha224(authcode.encode()).hexdigest().upper():
+			return line.split()[1]
+
+	return None
+
+
+def rfid_watcher():
+	global current_rfid_users
+	rfid_reader = evdev.InputDevice('/dev/input/event0')
+	print("Connected to RFID Reader")
```
Comment (lynxis) [inline mac_kicker.py:62]: use the logging framework?

File: mac_kicker.py [hunk 6]
```diff
@@ -16,38 +16,87 @@ def mac_tester():
...
 				if words[0] == "MAC":
 					mac_address = words[2].upper()
 					if mac_address in macs.keys():
-						current_users.append(macs[mac_address])
+						current_mac_users.append(macs[mac_address])
+
+		current_mac_users = list(set(current_mac_users)) # Dont duplicate users
+
+		# If the door is closed, kill all RFID users
+		if "LOCKED" in requests.get("http://door:8080/").text:
```
Comment (lynxis) [inline mac_kicker.py]: why not track the LOCK state always and only interact if UNLOCKED?

File: mac_kicker.py [hunk 7]
```diff
@@ -16,38 +16,87 @@ def mac_tester():
...
-		current_users = list(set(current_users)) # Dont duplicate users
 		time.sleep(60)
 
 
+def find_rfid_user(authcode):
+	with open("registered_rfid", "r") as f:
+		tokenlines = f.readlines()
+
+	for line in tokenlines:
+		if line.split()[0].upper() == hashlib.sha224(authcode.encode()).hexdigest().upper():
```
Comment (lynxis) [inline mac_kicker.py]: encode and hast it ones, not every line

File: mac_kicker.py [hunk 8]
```diff
@@ -16,38 +16,87 @@ def mac_tester():
...
+	for event in rfid_reader.read_loop():
+		if event.type == 1 and event.value == 1:
+			if event.code > len(keys):
+				continue
+
+			if keys[event.code] in "0123456789":
+				current_code += keys[event.code]
+			else:
+				rfid_user = find_rfid_user(current_code)
+				if rfid_user:
```
Comment (lynxis) [inline mac_kicker.py]: better use
<code>
if not rfid_user:
    continue

code.
</code>

**Review by lynxis — CHANGES_REQUESTED** [review]

File: rfid_reader.py (hunk 3 above)
Comment (c1ko) [reply rfid_reader.py:2]: requests is used...

File: mac_kicker.py (hunk 5 above)
Comment (c1ko) [reply mac_kicker.py:62]: Nope, as it does only log this single line

File: mac_kicker.py (hunk 6 above)
Comment (c1ko) [reply mac_kicker.py]: Weil es mehrere UNLOCKED states gibt. Aber das will ja keiner.

File: mac_kicker.py [hunk 9]
```diff
@@ -85,7 +96,7 @@ def on_message(self, target, source, message):
 		current_users = list(set(current_mac_users + current_rfid_users))
 		# don't respond to our own messages, as this leads to a positive feedback loop
 		if source != self.nickname:
-			if(".presence" in message or ".present" in message):
+			if (".presence" in message or ".present" in message):
```
Comment (lynxis) [inline mac_kicker.py:99]: don't use clamps here. They are not needed. I can tell you as c programmer.

**Comment by lynxis** [comment]
WTF is dead, nee

**Pull request closed without merging**